### DISCUSSION TITLE
Errors are displayed for the wrong fields when working with array fields

### DISCUSSION THREAD

**Discussion by ALIABUELRUB** [react-hook-form/react-hook-form#8173] [Q&A] [answered]
### Version Number

^7.29.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-array-fields-forked-vgwhfc?file=/src/index.js

### Steps to reproduce

1. Add a friend  (dont fill the fields)
2. hit submit (errors will be displayed for the required fields)
3. Add a new friend (the new friend will be added in the first row)
4. Errors of the first friend will be on the second friend and first friend will have no errors

### Expected behaviour

the errors of the first friend shouldn't be removed and second friend should have no errors because it is newly added

https://user-images.githubusercontent.com/55139883/162955482-c02f66b0-ac40-4279-8517-8e8287da0c8a.mp4

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
This is not an issue rather incorrect usage, clean up your codesandbox and remove unused code first. couple issues from observation: 

- `watch` is not made for retrieve value as such, rather subscribe form value and trigger re-render.
<img width="459" alt="Screen Shot 2022-04-12 at 10 05 59 pm" src="https://user-images.githubusercontent.com/10513364/162958934-090b2765-1e8e-4d09-bb63-2c1e51b0fb8d.png">

- there are field array action can be used instead `setValue`
<img width="607" alt="Screen Shot 2022-04-12 at 10 06 42 pm" src="https://user-images.githubusercontent.com/10513364/162959042-505f62a4-dba7-4c0c-a398-8a2582358bf6.png">

- should be using fields instead what's return by `watch`
<img width="641" alt="Screen Shot 2022-04-12 at 10 07 05 pm" src="https://user-images.githubusercontent.com/10513364/162959112-f8ca7e6d-9088-4e8d-becd-648d2cf18495.png">
 
please spend sometimes with the doc and follow the useFieldArray section

**Reply by ALIABUELRUB** [reply]
Thanks for the prompt reply.

Looks like I am miss using the library😅 will spend more time with the doc

Thanks again
